### PR TITLE
feat(server): Add support for ViewHierarchy attachment_type

### PR DESCRIPTION
**Pull request by markushi**
Add support for the upcoming [View Hierarchy](https://github.com/getsentry/rfcs/pull/33) feature.

If there's more work that needs to be done for relay, please let me know or kindly take over the PR 🙂 

I mapped the new attachment type to the String `"event.viewhierarchy"`, which matches with the existing `"event.applecrashreport"` - but I guess technically `"event.view_hierarchy"` would be more correct here.

### CODE REVIEW

File: relay-server/src/envelope.rs [hunk 1]
```diff
@@ -361,6 +361,9 @@ pub enum AttachmentType {
     /// [`symbolic_unreal::Unreal4LogEntry`]: https://docs.rs/symbolic/*/symbolic/unreal/struct.Unreal4LogEntry.html
     UnrealLogs,
 
+    /// An application ui view hierarchy (json payload)
```
Comment (iker-barriocanal) [inline relay-server/src/envelope.rs]:
```suggestion
    /// An application UI view hierarchy (json payload).
```

File: relay-server/src/envelope.rs [hunk 2]
```diff
@@ -382,6 +385,7 @@ impl fmt::Display for AttachmentType {
             AttachmentType::Breadcrumbs => write!(f, "event.breadcrumbs"),
             AttachmentType::UnrealContext => write!(f, "unreal.context"),
             AttachmentType::UnrealLogs => write!(f, "unreal.logs"),
+            AttachmentType::ViewHierarchy => write!(f, "event.viewhierarchy"),
```
Comment (iker-barriocanal) [inline relay-server/src/envelope.rs]: For the name, I'm leaning towards `"event.view_hierarchy"` but not strongly opinionated for the same reason you describe in the PR description. I'd match it with the type coming in the envelope though.

**Review by iker-barriocanal — APPROVED** [review]

**Review by jjbayer — APPROVED** [review]

File: relay-server/src/envelope.rs (hunk 2 above)
Comment (jjbayer) [reply relay-server/src/envelope.rs]: This was already discussed here: https://github.com/getsentry/rfcs/pull/33#discussion_r1032075538. But there seems to be a strong preference for `view_hierarchy` so go for it.

Comment (markushi) [reply relay-server/src/envelope.rs]: Ah I knew it was mentioned somewhere, but couldn't find it - thanks for linking!

**Pull request merged**